document/json/encoder.go

```diff
@@ -0,0 +1,318 @@
+package json
+
+import (
+	"fmt"
+	"math/big"
+	"reflect"
+
+	"github.com/aws/smithy-go/document"
+	"github.com/aws/smithy-go/document/internal/serde"
+	smithyjson "github.com/aws/smithy-go/encoding/json"
+)
+
+type EncoderOptions struct{}
+
+type Encoder struct {
```

Comment:
Why is this necessary? You appear to be using the standard library json marshaler anyway.

Reply:
Going to resolve this comment, see earlier response on the Decoder for the teams reasoning.